csdc-6/adcs-simulation/cpp/src/Simulator.cpp

```diff
@@ -30,6 +30,10 @@ Simulator::Simulator(Messenger *messenger)
     timestamp t(config.GetTimestepInMilliSeconds(),0);
     this->timestep_length = t;
     this->last_called = -1;
+
+    this->variableTimestep = config.GetTimestepDecision();

```

Comment:
Everything from line 28 -> line 36 should not be in Simulator.cpp.

These should be moved to the UI.cpp, in the function "get_sim_config". The parameters used should be added to the structure "sim_config".